profiler/src/main/java/com/splunk/opentelemetry/profiler/EventProcessingChain.java

```diff
@@ -204,32 +190,51 @@ public void close() {
    * change is detected process buffered events and clear the buffer.
    */
   private static class ChunkTracker {
-    private EventType contextAttachedEventType = null;
-    private EventType threadDumpEventType = null;
+    private final EventTypeHolder contextAttached = new EventTypeHolder();
```

Comment:
perhaps it would be better to place these in a map where key is event name so we wouldn't need to touch this again when we start consuming more events.